enhancements/installer/infrastructure-apis-custom-trust.md

```diff
@@ -0,0 +1,236 @@
+	//      -----END CERTIFICATE-----
+	//
+	// +optional
+	TrustedCA ConfigMapFileReference `json:"trustedCA,omitempty"`
+```
+
+### Cluster Network Operator: Proxy controller
+
+Currently the proxy config controller in CNO(cluster-network-operator) reads the [trustedCA][proxy-object-trusted-ca] from `Proxy` config object and generates a ConfigMap as detailed [here][egress-proxy-proposal] so that each operator can use the `config.openshift.io/inject-trusted-cabundle` injection to receive a trust bundle.
+
+The proxy config controller should,
+
+1. Add Infrastructure `.spec.trustedCA` as another source of input in addition to the `.spec.trustedCA` from Proxy object in creating the `openshift-config-managed/trusted-ca-bundle`.
+2. This makes it so that injection mechanism delivers the infrastructure trust bundle to all the operators using existing workflow.
+
+### Operators
+
+All the existing operators like [kube-controller-manger-operator][kcm-o-inject], [cloud-credential-operator][cco-inject], [cluster-ingress-operator][cio-inject], [cluster-registry-operator][cro-inject] already use the trust bundle injection to configure the trust in their containers and would need no change. But there are changes in two operators namely machine-api-operator and machine-config-operator to support the trust for infrastructure APIs.
+
+#### Machine API Operator
+
+The MAO(machine-api-operator) does not talk to infrastructure APIs itseld, rather it manages cluster-api providers per platform that communicate with these APIs. MAO should use the same process
+
+1. Create a ConfigMap in it's namespace that uses `config.openshift.io/inject-trusted-cabundle` to receive the trusted bundle, like [here][cro-inject].
+2. Mount the ConfigMap to appropriate location in the provider containers like [here][cio-mount-trust].
+
+#### Machine Config Operator
+
+MCO(machine-config-operator) is one of the few operators that cannot use the inject mechanism because it configures the `kubelet` on the host and we wouldn't want to the override the trust for the entire machine but add specific trust bundle for `kubelet`.
+
+There was previous work done to support a form of custom trust for [OpenStack][openstack-workaround-mco], but that is not widely implementable across other providers for various reasons mentioned [above][#Can-we-configure-the-custom-trust-for-SDKs-easily].
+
+On bootstrap host it reads the `<root>/tls/cloud-ca-cert.pem` for the custom trust, while in cluster it reads `ca-bundle.pem` key from ConfigMap specified in `.spec.cloudConfig`.
+
+Moving forward the MCO on bootstrap host will read the ConfigMap file `<root>/manifests/assets/manifests/infra-user-ca-bundle-config.yaml` for the trust bundle, and in cluster the operator will read the generated ConfigMap as described [above][#Infrastructure-API]. It should treat it similar to the Proxy trust bundle, i.e ensure it ends up in the list of machine trust [here][mco-additional-trust-list]
```

Comment:
I am missing how this is substantially different from what is done now, with respect to bootstrap?  both seem to be reading a file (in one case a .pem, in the other a config yaml that presumably leads to a .pem?)

Reply:
This makes the mco's resp converged to a ConfigMap on bootstrap (sourced from file) and in cluster (sourced from API)

Reply:
so fundamentally it's not substantially different.  it's an incremental move towards consistency, but no real technical implementation change or functional benefit.

(that's fine if true, just trying to make sure i'm not missing something about the limitations of the current implementation)